crates/api/src/values.rs

```diff
@@ -133,6 +133,13 @@ impl Val {
     pub fn unwrap_anyref(&self) -> AnyRef {
         self.anyref().expect("expected anyref")
     }
+
+    pub(crate) fn comes_from_same_store(&self, store: &Store) -> bool {
+        match self {
+            Val::FuncRef(f) => Store::same(store, f.store()),
+            _ => true,
```

Comment:
It'd be helpful to have a brief comment here, mentioning that none of the numeric value types inherently depend on a store.

Reply:
I went ahead and opted to switch this to an exhaustive match and added a comment for integers